pkg/utils/gpu.go

```diff
@@ -17,124 +17,81 @@ limitations under the License.
 package utils
 
 import (
+	_ "embed"
 	"strings"
+
+	"gopkg.in/yaml.v2"
 )
 
 // TODO: Get these from agentbaker
 const (
-	Nvidia470CudaDriverVersion = "cuda-470.82.01"
-	Nvidia550CudaDriverVersion = "cuda-550.54.15"
-	Nvidia535GridDriverVersion = "grid-535.161.08"
-
-	// These SHAs will change once we update aks-gpu images in aks-gpu repository. We do that fairly rarely at this time.
-	// So for now these will be kept here like this and periodically bump them
-	AKSGPUGridSHA = "sha-d1f0ca"
-	AKSGPUCudaSHA = "sha-2d4c96"
+	Nvidia470CudaDriverVersion = "470.82.01"
+
+	// https://github.com/Azure/AgentBaker/blob/ddf36a24eafd02ce0589657ff2dc799125f4ad37/parts/linux/cloud-init/artifacts/components.json#L562
+	NvidiaCudaDriverVersion = "550.90.12"
+	AKSGPUCudaVersionSuffix = "20241021235610"
+
+	NvidiaGridDriverVersion = "535.161.08"
+	AKSGPUGridVersionSuffix = "20241021235607"
 )
 
-func GetAKSGPUImageSHA(size string) string {
-	if UseGridDrivers(size) {
-		return AKSGPUGridSHA
-	}
-	return AKSGPUCudaSHA
+type NvidiaSKUConfig struct {
+	NvidiaEnabledSKUFamilies        map[string][]string `yaml:"nvidiaEnabledSKUs"`
```

Comment:
this is a structure of 

[family][]string{sku_sizea, sku_sizeb}, but we don't use family directly. can we write a marshaler to take in the final structure we want? 

map[sku_sizea] = supported_boolean?

Reply:
See elsewhere on the value of alignment with upstream